### PR TITLE
check if lhs of property access expression is a module

### PR DESCRIPTION
**Pull request by dariajung**
Add a check to see if the Left Hand Side expression is a module. 

Needs a unit test, and E2E test.

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -89,6 +92,12 @@ export class Minifier {
 
     // given a base path, preserve file directory structure
     var subFilePath = filePath.replace(this._minifierOptions.basePath, '');
+
+    if (subFilePath === filePath) {
+      filePath = path.relative(process.cwd(), filePath);
```
Comment (rkirov) [inline src/main.ts]: doesn't look right, you already normalized on line 84.

File: src/main.ts [hunk 2]
```diff
@@ -105,6 +114,13 @@ export class Minifier {
         output += this.visit(pae.expression);
         output += pae.dotToken.getText();
 
+        // if LHS is a module, do not rename property name
+        var lhsIsModule = false;
+        var lhsTypeSymbol = this._typeChecker.getTypeAtLocation(pae.expression).symbol;
+        if (lhsTypeSymbol) {
```
Comment (rkirov) [inline src/main.ts]: would read better as 

```
if (lhsTypeSymbol && ts.SymbolFlags.ValueModule === lhsTypeSymbol.flags) {
   lhsIsModule = false
}
```

File: src/main.ts [hunk 3]
```diff
@@ -89,6 +92,12 @@ export class Minifier {
 
     // given a base path, preserve file directory structure
     var subFilePath = filePath.replace(this._minifierOptions.basePath, '');
```
Comment (rkirov) [inline src/main.ts]: make sure basePath is absolute, otherwise this would be a bug.

Comment (dariajung) [inline src/main.ts]: Line 91 makes it absolute: `this._minifierOptions.basePath = path.resolve(process.cwd(), this._minifierOptions.basePath);`

**Comment by dariajung** [comment]
@rkirov Please take another look.

File: test/input/b.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+import a = require('./a');
+var foo = new a.Foo();
+foo.bar = 'bar';
+console.log('This should be bar: ' + foo.bar);
```
Comment (rkirov) [inline test/input/b.ts]: This should be an assertion.

**Comment by rkirov** [comment]
LGTM, add the assertion and submit it.

File: test/input/b.ts (hunk 4 above)
Comment (dariajung) [inline test/input/b.ts]: changed to assertion

**Comment by dariajung** [comment]
Merged

**Pull request closed without merging**